service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java

```diff
@@ -25,42 +25,49 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(PayonePostServiceImpl.class);
     private static final String ENCODING_UTF8 = "UTF-8";
+    private final List<String> personalDataToRemove;
 
     private String serverAPIURL;
 
     private int connectionTimeout = 3000;
 
 
-    private PayonePostServiceImpl(final String serverAPIURL) {
+    private PayonePostServiceImpl(final String serverAPIURL,
+                                  final List<String> personalDataToRemove) {
         Preconditions.checkArgument(
                 serverAPIURL != null && !serverAPIURL.isEmpty(),
                 "The server api url must not be null or empty.");
+        Preconditions.checkArgument(personalDataToRemove != null,
+                "personalDataToRemove cannot be null");
         this.serverAPIURL = serverAPIURL;
+        this.personalDataToRemove = personalDataToRemove;
     }
 
     /**
      * Initialize new service with url.
      *
      * @param payoneServerApiUrl - the payone server api url, must not be null or empty
+     * @param personalDataToRemove - list of parameters that should be removed from the logs
```

Comment:
Why one has to define full list of parameters to be removed? Isn't it obvious which data qualifies as personal and which not?

Reply:
It's just more flexible this way